src/sentry/runner/settings.py

```diff
@@ -79,6 +80,22 @@ def discover_configs():
     )
 
 
+_warning_logger = logging.getLogger("sentry.warnings")
+
+
+# This is set to warnings.showwarning to intercept all warnings and report them to Sentry.
+# The python SDK doesn't have a warnings integration at the time of writing,
+# and capture_message is very limited, so we just log an error here since LOGGING setting
+# is already configured to send error logs to the SDK's EventHandler.
+def _showwarning(message, category, filename, lineno, file=None, line=None):
+    _warning_logger.error(
```

Comment:
woudl wrap this in `safe_execute` and also call the previous value of `warnings.showwarning`

Reply:
ok, no safe_execute necessary I guess

would still call the old showwarning here